docs/topics/classes.md

```diff
@@ -1,245 +1,692 @@
+    println("Person object created: ${person1.name}")
+    // Person object created: Sebastian
+
+    // Creates an object by passing a specific value
+    val person2 = Person("Joe")
+    println("Person object created: ${person2.name}")
+    // Person object created: Joe
+}
+```
+{kotlin-runnable="true"}
+
+> In Kotlin, unlike other object-oriented programming languages like Java, 
+> there is no need for the `new` keyword when creating objects of a class.
+>
+{style="note"}
+
+Additionally, you can create objects inside another function and call that function from `main()`.
+
+For information about creating objects of nested, inner, and anonymous inner classes,
+see the [Nested classes](nested-classes.md) section.
 
 ## Inheritance
 
-Classes can be derived from each other and form inheritance hierarchies.
-[Learn more about inheritance in Kotlin](inheritance.md).
+Class inheritance in Kotlin allows you to create a new class (derived class) from an existing class (base class), 
+inheriting its properties and functions while adding or modifying behavior.
+
+> For detailed information about inheritance hierarchies and the use of the `open` keyword, see the [Inheritance](inheritance.md) section.
+> 
+{style="note"}
 
 ## Abstract classes
 
-A class may be declared `abstract`, along with some or all of its members.
-An abstract member does not have an implementation in its class.
-You don't need to annotate abstract classes or functions with `open`.
+An abstract class in Kotlin is a class that cannot be instantiated directly and is designed to be inherited 
+by other classes. 
+
+An abstract class can define abstract properties and functions, which must be implemented 
+by subclasses.
+
+You can declare an abstract class using the `abstract` modifier:
 
 ```kotlin
-abstract class Polygon {
-    abstract fun draw()
-}
+abstract class Person
+```
 
-class Rectangle : Polygon() {
-    override fun draw() {
-        // draw the rectangle
-    }
-}
+Like any other class, abstract classes can also have constructors. 
+These constructors initialize class properties and enforce required parameters for subclasses:
+
+```kotlin
+abstract class Person(val name: String, val age: Int)
 ```
 
-You can override a non-abstract `open` member with an abstract one.
+An abstract class can have both abstract and non-abstract members (properties and functions). 
+To declare a member as abstract, you must use the `abstract` keyword explicitly.
+
+Abstract members do not have an implementation in the abstract class. The `override`
+keyword ensures that a subclass provides an actual implementation for an abstract function:
 
 ```kotlin
-open class Polygon {
-    open fun draw() {
-        // some default polygon drawing method
+// Abstract class with a primary constructor that defines 'name' and 'age'
+abstract class Person(
+    val name: String,
+    val age: Int
+) {
+    // Abstract member (doesn't provide default implementation, must be implemented by subclasses)
```

Comment:
```suggestion
    // Abstract member (doesn't provide implementation, must be implemented by subclasses)
```

Reply:
It's a bit more precise to talk about base (not default) implementations, as "default" term is already used in Kotlin for other things.